lib/IRGen/GenStruct.cpp

```diff
@@ -250,6 +250,18 @@ namespace {
       return fields[0].getTypeInfo().isSingleRetainablePointer(expansion, rc);
     }
 
+    void destroy(IRGenFunction &IGF, Address address, SILType T,
+                 bool isOutlined) const override {
+      // If the struct has a deinit declared, then call it to destroy the
+      // value.
+      if (tryEmitDestroyUsingDeinit(IGF, address, T)) {
+        return;
+      }
```

Comment:
Nice. Does this mean we can delete the `SILMoveOnlyDeinitInsertionPass` ?

Reply:
It still seems like a useful optimization to have, since devirtualizing the `destroy_value` allows to possibly inline and/or specialize the deinit. But it shouldn't be mandatory to run it.